### PR TITLE
Let generator add RequiredPermission-decorator

### PR DESCRIPTION
**Pull request by fraxachun**
Depends on https://github.com/vivid-planet/comet/pull/1430

### CODE REVIEW

**Review by nsams — DISMISSED** [review]
product.entity.ts is missing I think?

**Comment by fraxachun** [comment]
> product.entity.ts is missing I think?

The generator automatically adds `@RequiredPermission([instanceNamePlural])`, so for `product.entity.ts` it's correct when the permission should be products. However, for `product-category` and `product-tag` it has to be overwritten.

File: packages/api/cms-api/src/generator/generate-crud.ts [hunk 1]
```diff
@@ -701,11 +704,15 @@ function generateResolver({ generatorOptions, metadata }: { generatorOptions: Cr
         imports.push(generateEntityImport(scopeProp.targetMeta, generatorOptions.targetDirectory));
     }
 
+    if (instanceNamePlural === "productCategories") {
+        console.log(generatorOptions);
+    }
+
```
[the remark this answers is not in the document]
Comment (fraxachun) [reply packages/api/cms-api/src/generator/generate-crud.ts]: ffa52c2861b049d8782f2364b7b4241c279c6bc6

File: packages/api/cms-api/src/generator/crud-generator.decorator.ts [hunk 2]
```diff
@@ -1,24 +1,27 @@
 export interface CrudGeneratorOptions {
     targetDirectory: string;
+    requiredPermission?: string;
```
[the remark this answers is not in the document]
Comment (fraxachun) [reply packages/api/cms-api/src/generator/crud-generator.decorator.ts]: Yes, better this way. 061c098414dfd5f1a25c73102f3278918bc9d1d9

File: packages/api/cms-api/src/generator/crud-generator.decorator.ts [hunk 3]
```diff
@@ -1,24 +1,27 @@
 export interface CrudGeneratorOptions {
     targetDirectory: string;
+    requiredPermission?: [string];
```
Comment (nsams) [inline packages/api/cms-api/src/generator/crud-generator.decorator.ts:3]: `[string] | string` would be appropriate here, as in most cases you'll use a single string and the option is named in singular form

**Review by nsams — APPROVED** [review]

**Pull request merged**